library/src/main/java/com/novoda/downloadmanager/lib/DownloadManager.java

```diff
@@ -378,30 +378,50 @@
     private final ContentResolver contentResolver;
     private final DownloadsUriProvider downloadsUriProvider;
     private final SystemFacade systemFacade;
+    private final BatchRepository batchRepository;
 
     private Uri baseUri;
 
-    public DownloadManager(Context context, ContentResolver resolver) {
-        this(context, resolver, DownloadsUriProvider.getInstance(), new RealSystemFacade(context), false);
+    public static DownloadManager newInstance(Context context, ContentResolver contentResolver) {
```

Comment:
as @amlcurran said it's nice keeping the client away from our own patterns, but at the same time we've done this before (https://github.com/novoda/notils/blob/master/src/main/java/com/novoda/notils/logger/toast/Toaster.java#L28-L30) so I'm not super bothered